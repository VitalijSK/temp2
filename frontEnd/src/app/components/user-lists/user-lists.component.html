<div class="list">
  <app-user-element *ngFor="let user of users; trackBy: trackByFn" [name]="user.name" [id]="user.id" (deleteId)="deleteUserById($event)" (selectId)="selectUserById($event)"></app-user-element>
</div>
