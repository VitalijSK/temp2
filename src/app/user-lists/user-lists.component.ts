import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
import { IUser } from 'server/models/User';

@Component({
  selector: 'app-user-lists',
  templateUrl: './user-lists.component.html',
  styleUrls: ['./user-lists.component.scss']
})
export class UserListsComponent implements OnInit {
  
  @Input() users: IUser[];
  @Output() selectUser : EventEmitter<IUser> = new EventEmitter<IUser>();
  @Output() deleteUser : EventEmitter<string> = new EventEmitter<string>();

  constructor() { }

  ngOnInit() {
  }
  
  selectUserById(id: string) {
    const user = this.users.find(userById(id));
    this.selectUser.emit(user);
  }
  deleteUserById(id: string) {
    this.deleteUser.emit(id);
  }
  trackByFn(index : number, user : IUser) {
    return user.id;
  }
}

const userById = (id : string) => (user : IUser) => user.id === id;
