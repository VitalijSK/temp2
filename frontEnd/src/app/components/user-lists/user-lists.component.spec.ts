import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { UserListsComponent } from './user-lists.component';

describe('UserListsComponent', () => {
  let component: UserListsComponent;
  let fixture: ComponentFixture<UserListsComponent>;


  it('renders markup to snapshot', () => {
    expect('fixture').toBeTruthy();
  });
});
