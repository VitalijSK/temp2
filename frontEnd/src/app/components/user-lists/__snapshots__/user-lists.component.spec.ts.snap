// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserListsComponent renders markup to snapshot 1`] = `undefined`;
